Added final results encryption option

hugoseabra-imachines opened the pull request:
Added flag to decide whether final results in Job Bulk payouts must be encrypted or plain;

hugoseabra committed:
feat: adding encryption module with better state control and explicit approach

Encryption now has a dedicated class with explicit Algorithm approach.

IMI-2282

hugoseabra committed:
feat: adding encryption module with better state control and explicit approach

Encryption now has a dedicated class with explicit Algorithm approach.

IMI-2282

hugoseabra committed:
feat: eth_bridge with tests in dedicated folder

IMI-2282

hugoseabra committed:
feat: adding option to upload data encrypted or plain

IMI-2282

hugoseabra committed:
feat: bulk payout now with option to encrypt final results

Final results can be plain or encrypted with new parameter um `bulk_payout` method.

IMI-2282

hugoseabra committed:
chore: changing python tests command

IMI-2282

hugoseabra committed:
test: correction in abort test

Returnign `tx` constant.

IMI-2282

hugoseabra committed:
test: moving tests to dedicated repository

IMI-2282

hugoseabra committed:
ref: crypto refactored with dedicated module and tests

IMI-2282

hugoseabra committed:
Merge branch 'refactor/crypto-refactoring-with-tests' into feature/crypto-refactoring-with-tests

# Conflicts:
#	hmt_escrow/crypto/__init__.py
#	hmt_escrow/crypto/encryption.py
#	hmt_escrow/eth_bridge.py
#	hmt_escrow/job.py
#	hmt_escrow/storage.py
#	test/hmt_escrow/test_eth_bridge.py
#	test/hmt_escrow/test_job.py
#	test/hmt_escrow/test_storage.py
#	test/hmt_escrow/utils/__init__.py

hugoseabra committed:
feat: Adding support to decide whether final results must be encrypted

IMI-2282

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/humanprotocol/hmt-escrow?pullRequest=330) <br/>All committers have signed the CLA.

hugoseabra-imachines committed:
Merge branch 'master' into feature/adding-final-results-encryption-option

# Conflicts:
#	bin/shell.sh
#	hmt_escrow/job.py
#	hmt_escrow/storage.py
#	test/hmt_escrow/test_job.py
#	test/hmt_escrow/test_storage.py

hugoseabra-imachines committed:
chore: applying lint

IMI-2282

hugoseabra-imachines committed:
fix: handling s3 content wth correct type

Content that comes from s3 is bytes-like, but decrypting and delivering retrieved content must be converted to string to be turned to dict with json dumps.

IMI-2282

hugoseabra-imachines committed:
test: adding comments for encryption option tests

Better understanding of tests.

IMI-2282

hugoseabra-imachines committed:
chore: bumping version to 0.10.3

Better understanding of tests.

IMI-2282

hugoseabra-imachines committed:
Merge branch 'master' into feature/adding-final-results-encryption-option

# Conflicts:
#	hmt_escrow/crypto/encryption.py
#	hmt_escrow/storage.py
#	package-lock.json
#	package.json
#	setup.py

vkomodey commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "hmt-escrow",
-  "version": "0.10.2",
+  "version": "0.10.3",
```
This is a compatible API change. The version should be 0.11.0

hugoseabra-imachines replied:
@vkomodey , I kept usage compatibility with upload by adding optional flag parameter. I really believe it is a minor version as it does not break other consumers. Don't you think we should keep minor change in this way?

vkomodey replied:
According to [semver](https://semver.org/) rules, we should change the PATCH number:

> * MAJOR version when you make incompatible API changes,
> * MINOR version when you add functionality in a backwards compatible manner, and
> * PATCH version when you make backwards compatible bug fixes.

hugoseabra-imachines replied:
Oh sure! Doing it!

vkomodey reviewed (changes requested):
lgtm

vkomodey commented:
And please, squash all your commits into 2 or 3 or 4 commits. Thanks

hugoseabra-imachines committed:
chore: bumping version to 0.11.0

IMI-2282

hugoseabra-imachines commented:
> And please, squash all your commits into 2 or 3 or 4 commits. Thanks

Due to the sequence of commits where there are commits that are before the merge with `master`, when squashing the commits, the changes before `master` is lost.

Is it possible to merge this without squashing?

The sequence of commits. Just note that there commits of this branch before `master` merge:
![image](https://user-images.githubusercontent.com/106671365/175135770-cc468492-63fd-4b27-8e99-d559c9189210.png)

vkomodey commented:
@hugoseabra-imachines 
Please, never use `merge ${stable branch name, i.e. master}` to your dev branches. To keep your commit history clean - please, always use the `rebase` command. 

`git pull origin master --rebase`

It will allow you to keep master commits behind new commits from your branch.

As for this branch, you can check out a new branch(from master) and do a cherry-pick of all required commits into your history. Or simple copy&paste content from this PR into the new branch will do the work. I would say it is the easiest and quickest option here

igervasiychuk commented:
will this do? @vkomodey  https://github.com/humanprotocol/hmt-escrow/pull/336

vkomodey commented:
Duplicated and merged in #336
